apps/content-publishing-api/src/controllers/v3/content.controller.v3.ts

```diff
@@ -84,45 +109,82 @@ export class ContentControllerV3 {
         if (fileIndex === 0) {
           throw new BadRequestException('No files provided in the request');
         }
-
-        // Validate schema IDs match file count
         if (schemaIds.length !== fileProcessingPromises.length) {
           throw new BadRequestException('Number of schema IDs does not match number of files');
         }
 
-        // Process uploaded files
-        const uploadResults = await Promise.all(fileProcessingPromises);
+        const uploadResults = await Promise.allSettled(fileProcessingPromises);
+
+        const responseFiles: { referenceId?: string; cid?: string; error?: string }[] = [];
+        let hasFailedUploads = false;
+        const successfulUploads: { uploadResult: IFileResponse; originalIndex: number }[] = [];
+
+        uploadResults.forEach((uploadResult, index) => {
+          if (uploadResult.status === 'fulfilled' && uploadResult.value.cid && !uploadResult.value.error) {
+            successfulUploads.push({ uploadResult: uploadResult.value, originalIndex: index });
+            // Placeholder - will be filled after batch creation
+            responseFiles.push({ cid: uploadResult.value.cid });
+          } else {
+            hasFailedUploads = true;
+            const error =
+              uploadResult.status === 'rejected'
+                ? uploadResult.reason?.message || 'Upload failed'
+                : uploadResult.value.error || 'Upload failed';
+            responseFiles.push({ error });
+          }
+        });
 
-        // Check for upload errors
-        const errors = uploadResults.filter((uploadResult) => uploadResult.error);
-        if (errors.length > 0) {
-          throw new BadRequestException(errors[0].error || 'File upload failed');
+        // Create batch announcements for successful uploads
+        if (successfulUploads.length > 0) {
```

Comment:
Discussion point: is it better to take an all/nothing approach here; ie, if there are any upload failures, skip the batch announcements completely? I don't think I have a strong opinion here either way, but probably worth discussing.

Reply:
Agree that respond with upload status of each item in the batch.  I'm not sure of the value of skipping batch announcements entirely if some are successful.  As long as the reponse can be parsed in order to try again and debug.

Reply:
Let's table this for now and have an in-person whiteboarding discussion on it to decide.